contracts/L2_TokenRetriever.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import "./Lockable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@uma/core/contracts/common/implementation/MultiCaller.sol";
+import { WithdrawalAdapter } from "./chain-adapters/l2/WithdrawalAdapter.sol";
+
+interface IBridgeAdapter {
+    function withdrawToken(
+        address recipient,
+        uint256 amountToReturn,
+        address l2Token
+    ) external;
+
+    function withdrawTokens(WithdrawalAdapter.WithdrawalInformation[] memory) external;
+}
+
+/**
+ * @notice Contract deployed on an arbitrary L2 to act as an intermediate contract for withdrawals from L3 to L1.
+ * @dev Since each network has its own bridging requirements, this contract delegates that logic to a bridge adapter contract
+ * which performs the necessary withdraw action.
+ */
+contract L2_TokenRetriever is Lockable {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    // Should be set to the bridge adapter which contains the proper logic to withdraw tokens on
+    // the deployed L2
+    address public immutable bridgeAdapter;
+    // Should be set to the L1 address which will receive withdrawn tokens.
+    address public immutable tokenRetriever;
```

Comment:
If this is the address that should be set as the recipient of tokens on the L1 then `tokenRetriever` is a bit counter-intuitive for me. Would `tokenRecipient` be more descriptive?

```suggestion
    address public immutable tokenRecipient;
```

Reply:
I do think so. I had the blast and polygon contracts on the mind, but it's nice to not repeat retriever when we don't have to. https://github.com/across-protocol/contracts/pull/605/commits/d3f624bff2d3b263be3a310a5c9ef36a19075bc0